### PR TITLE
[MACOS] Fix two compiler errors by changing syntax.

### PR DESCRIPTION
**Pull request by vinhig**
Diving into the code and trying to compile it with pre-built tools binaries gave me two different errors:

```
os/lib/rtl/base/build/../fp2int.c:1326:20: error: bitwise negation of a boolean expression; did you mean logical negation? [-Werror,-Wbool-operation]
```
_Fixed by applying a little if_
```
os/apps/swiss/uos/../swlib/pattern.c:906:42: error: adding 'int' to a string does not append to the string [-Werror,-Wstring-plus-int]
```
_Fixed by replacing a pointer offset by pointer of an array element_

So before implementing bigger features, I wanted to make sure it compiles on my machine. Here is the change. Enjoy!

### CODE REVIEW

**Comment by evangreen** [comment]
Thanks Vincent. Would you be willing to sign the CAA [here](https://www.minocacorp.com/ca/individual/)? Then I can review and pull your change.

**Comment by vinhig** [comment]
I've just signed it. We can proceed to the next part.

File: lib/rtl/base/fp2int.c [hunk 1]
```diff
@@ -1323,7 +1323,10 @@ Return Value:
     }
 
     Significand = (Significand + RoundIncrement) >> 10;
-    Significand &= ~(((RoundBits ^ 0x200) == 0) & (RoundNearestEven != FALSE));
+    if (((RoundBits ^ 0x200) == 0) && (RoundNearestEven != FALSE)) {
+        Significand &= ~1;
+    }
+    // Significand &= ~(((RoundBits ^ 0x200) == 0) & (RoundNearestEven != FALSE));
```
Comment (evangreen) [inline lib/rtl/base/fp2int.c:1329]: Delete the commented out code and replace with a blank line

**Comment by evangreen** [comment]
Thanks! I left one minor comment, with that resolved this should be good to merge.

**Comment by vinhig** [comment]
Code has been modified according to review.

**Comment by evangreen** [comment]
Looks good. I squashed the commits, added some more description to the commit message, and pushed this as 6750d31e02d4f6f745a88caa4dd05f2f672b8832. Thanks!

**Pull request closed without merging**